src/History/EmailSubscriptionHistoryHandler.php

```diff
@@ -90,4 +93,24 @@ private function createEmailSubscriptionHistory(
             }
         }
     }
+
+    public function queryCountByMonth(Adherent $referent, int $months = 6): array
+    {
+        foreach (range(0, $months - 1) as $month) {
+            $until = (new Carbon("last day of -$month month"))->setTime(23, 59, 59, 999);
```

Comment:
what is the advantage of Carbon here instead of \DateTime class?

Reply:
It allows us to freeze/mock the clock/time in our test. We cannot do that if we use \DateTime directly.

For example in Behat or Phpunit we can set the clock date to 01/01/2018 so that when we call this method we would have always the same last 6 months:

- 01/2018
- 12/2017
- 11/2017
- etc.

Reply:
I see, thanks 👍